openstf/pipeline/train_model.py

```diff
@@ -135,15 +135,18 @@ def train_model_pipeline_core(
 
     # Split data
     train_data, validation_data, test_data = split_data_train_validation_test(
-        data_with_features.sort_index(axis=1)
+        data_with_features.sort_index(axis=0)
```

Comment:
We seem to still quite a few of these `sort_index` calls. This raises similar question as with the `load` and `horizon` columns. Why do we need to do this, when is this required, why is the data not in the right format in the first place, is it really the callers responsibility. It seems a bit random and messy right now?